lms/djangoapps/courseware/date_summary.py

```diff
@@ -240,7 +240,11 @@ class CourseStartDate(DateSummary):
 
     @property
     def date(self):
-        return self.course.start
+        enrollment = CourseEnrollment.get_enrollment(self.user, self.course_id)
```

Comment:
Quick question, is this supposed to be the default logic all of the time, or just for the new view? I ask since this summary class is pretty old, not sure where else it's used, as well as we're not adding an additional query every time we reference this field, could be worth just doing it once on instantiation of the class so each time we call .date we aren't sending another query out?

Reply:
I looked around for other uses of this class and I only found it being used for displaying the start date on the Dates tab so it felt safe to put this logic here. I wasn't sure if it would be better to create a class called CourseEnrollmentDate and then choose which one to use (CourseEnrollmentDate vs CourseStartDate) within `get_course_date_blocks` in `courses.py`...

Re: querying every time we call .date -- I'm gonna go with Mike's suggestion of early exiting if the course isn't self paced to avoid the unnecessary DB call.